src/HDF5.jl

```diff
@@ -1415,6 +1415,9 @@ function d_write(parent::Union{File,Group}, name::String, data; pv...)
     obj, dtype = d_create(parent, name, data; pv...)
```

Comment:
We should probably also add something like the following (perhaps ArgumentError)  here and below
```suggestion
    haskey(parent, name) && error( name, " already exists")
```

to prevent errors that get thrown like the following
```julia
julia> hid = h5open(tempname(), "w")
HDF5 data file: C:\Users\Mus\AppData\Local\Temp\jl_zDsvcXjFU3

julia> write(hid, "A", rand(3,3))

julia> write(hid, "A", rand(3,3))
HDF5-DIAG: Error detected in HDF5 (1.10.5) thread 0:
  #000: E:/mingwbuild/mingw-w64-hdf5/src/hdf5-1.10.5/src/H5D.c line 145 in H5Dcreate2(): unable to create dataset
    major: Dataset
    minor: Unable to initialize object
  #001: E:/mingwbuild/mingw-w64-hdf5/src/hdf5-1.10.5/src/H5Dint.c line 329 in H5D__create_named(): unable to create and link to dataset
    major: Dataset
    minor: Unable to initialize object
  #002: E:/mingwbuild/mingw-w64-hdf5/src/hdf5-1.10.5/src/H5L.c line 1557 in H5L_link_object(): unable to create new link to object
    major: Links
    minor: Unable to initialize object
  #003: E:/mingwbuild/mingw-w64-hdf5/src/hdf5-1.10.5/src/H5L.c line 1798 in H5L__create_real(): can't insert link
    major: Links
    minor: Unable to insert object
  #004: E:/mingwbuild/mingw-w64-hdf5/src/hdf5-1.10.5/src/H5Gtraverse.c line 851 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #005: E:/mingwbuild/mingw-w64-hdf5/src/hdf5-1.10.5/src/H5Gtraverse.c line 627 in H5G__traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #006: E:/mingwbuild/mingw-w64-hdf5/src/hdf5-1.10.5/src/H5L.c line 1593 in H5L__link_cb(): name already exists
    major: Links
    minor: Object already exists
ERROR: Error creating dataset //A
Stacktrace:
  [1] error(::String, ::String, ::String, ::String)
    @ Base .\error.jl:42
  [2] h5d_create
    @ ~\.julia\dev\HDF5\src\api.jl:158 [inlined]
  [3] d_create(parent::HDF5.File, path::String, dtype::HDF5.Datatype, dspace::HDF5.Dataspace; pv::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ HDF5 ~\.julia\dev\HDF5\src\HDF5.jl:733
  [4] d_create
    @ ~\.julia\dev\HDF5\src\HDF5.jl:730 [inlined]
  [5] d_create(parent::HDF5.File, name::String, data::Matrix{Float64}; pv::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ HDF5 ~\.julia\dev\HDF5\src\HDF5.jl:1405
  [6] d_create
    @ ~\.julia\dev\HDF5\src\HDF5.jl:1402 [inlined]
  [7] d_write(parent::HDF5.File, name::String, data::Matrix{Float64}; pv::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
    @ HDF5 ~\.julia\dev\HDF5\src\HDF5.jl:1424
  [8] d_write
    @ ~\.julia\dev\HDF5\src\HDF5.jl:1424 [inlined]
  [9] #write#45
    @ ~\.julia\dev\HDF5\src\HDF5.jl:1470 [inlined]
 [10] write(parent::HDF5.File, name::String, data::Matrix{Float64})
    @ HDF5 ~\.julia\dev\HDF5\src\HDF5.jl:1470
 [11] top-level scope
    @ REPL[3]:1


```

Reply:
Although this is probably  best left to a separate PR (I think this is systemic with other methods, that I haven't fully looked at, when called on twice).

Reply:
E.g. all the high level create/write methods also suffer from the same issue.

Reply:
Yeah, I think that'll be part of the method cleanup that needs to happen.